delta-lake/delta-spark321db/src/main/scala/com/databricks/sql/transaction/tahoe/rapids/GpuDoAutoCompaction.scala

```diff
@@ -32,13 +32,16 @@ import org.apache.spark.sql.SparkSession
 object GpuDoAutoCompaction extends PostCommitHook
     with DeltaLogging
     with Serializable {
-  override val name: String = "GpuDoAutoCompaction"
+  override val name: String = "Triggers compaction if necessary"
 
   override def run(spark: SparkSession,
                    txn: OptimisticTransactionImpl,
                    committedActions: Seq[Action]): Unit = {
     val gpuTxn = txn.asInstanceOf[GpuOptimisticTransaction]
     val newTxn = new GpuDeltaLog(gpuTxn.deltaLog, gpuTxn.rapidsConf).startTransaction()
+    // Note: The Databricks AutoCompact PostCommitHook cannot be used here
+    // (with a GpuOptimisticTransaction). It does not appear to use OptimisticTransaction.writeFiles
+    // to write the compacted file.
```

Comment:
I suspect it does use writeFiles, and the issue is that it creates a new transaction, which causes us to "lose" the GpuOptimisticTransaction and thus the override of writeFiles.

Reply:
I'll reword the code comments here to reflect that.

Reply:
Clarified.